cloud_uls_factory assumes that all cloud clients are provided in CloudManager

`CloudManager` is set to contain clients for cloud providers that are set in `AUTH`:
https://github.com/red-hat-storage/ocs-ci/blob/4bf2986e037540c3fba363363a0a57f1332bf655/ocs_ci/ocs/resources/cloud_manager.py#L44

`cloud_uls_factory` expects that all clients are created (also those that are not in `AUTH`):
https://github.com/red-hat-storage/ocs-ci/blob/e1632d6c6c134e37af7e7737960f0b2e69aeff1d/ocs_ci/ocs/resources/cloud_uls.py#L29

This leads to error when some credential in AUTH is missing:
```
(...)
request = <SubRequest 'cloud_uls_factory' for <Function test_namespace_resource_creation>>
cld_mgr = <ocs_ci.ocs.resources.cloud_manager.CloudManager object at 0x7ff9acb66128>

    def cloud_uls_factory(request, cld_mgr):
        """
        Create an Underlying Storage factory.
        Calling this fixture creates a new underlying storage(s).
    
        Args:
            request (object): Pytest built-in fixture
            cld_mgr (CloudManager): Cloud Manager object containing all connections to clouds
    
        Returns:
            func: Factory method - each call to this function creates
                an Underlying Storage factory
    
        """
        all_created_uls = {
            'aws': set(),
            'google': set(),
            'azure': set(),
            'ibmcos': set()
        }
    
        ulsMap = {
            'aws': cld_mgr.aws_client,
>           'google': cld_mgr.google_client,
            'azure': cld_mgr.azure_client,
            # TODO: Implement - 'ibmcos': cld_mgr.ibmcos_client
        }
E       AttributeError: 'CloudManager' object has no attribute 'google_client'

```

--- a/ocs_ci/ocs/bucket_utils.py
+++ b/ocs_ci/ocs/bucket_utils.py
@@ -297,6 +297,16 @@ def cli_create_aws_backingstore(mcg_obj_session, cld_mgr, backingstore_name, uls
 
 
 def oc_create_google_backingstore(cld_mgr, backingstore_name, uls_name, region):
+    """
+    Create a new backingstore with GCP underlying storage using oc create command
+
+    Args:
+        cld_mgr (CloudManager): holds secret for backingstore creation
+        backingstore_name (str): backingstore name
+        uls_name (str): underlying storage name
+        region (str): which region to create backingstore (should be the same as uls)
+
+    """
     bs_data = templating.load_yaml(constants.MCG_BACKINGSTORE_YAML)
     bs_data['metadata']['name'] = backingstore_name
     bs_data['spec'] = {
@@ -331,6 +341,16 @@ def cli_create_google_backingstore(mcg_obj_session, cld_mgr, backingstore_name, 
 
 
 def oc_create_azure_backingstore(cld_mgr, backingstore_name, uls_name, region):
+    """
+    Create a new backingstore with Azure underlying storage using oc create command
+
+    Args:
+        cld_mgr (CloudManager): holds secret for backingstore creation
+        backingstore_name (str): backingstore name
+        uls_name (str): underlying storage name
+        region (str): which region to create backingstore (should be the same as uls)
+
+    """
     bs_data = templating.load_yaml(constants.MCG_BACKINGSTORE_YAML)
     bs_data['metadata']['name'] = backingstore_name
     bs_data['spec'] = {

--- a/ocs_ci/ocs/bucket_utils.py
+++ b/ocs_ci/ocs/bucket_utils.py
@@ -263,9 +263,16 @@ def oc_create_aws_backingstore(cld_mgr, backingstore_name, uls_name, region):
     bs_data = templating.load_yaml(constants.MCG_BACKINGSTORE_YAML)
     bs_data['metadata']['name'] = backingstore_name
     bs_data['metadata']['namespace'] = config.ENV_DATA['cluster_namespace']
-    bs_data['spec']['awsS3']['secret']['name'] = cld_mgr.aws_client.secret.name
-    bs_data['spec']['awsS3']['targetBucket'] = uls_name
-    bs_data['spec']['awsS3']['region'] = region
+    bs_data['spec'] = {
+        'type': 'aws-s3',
+        'awsS3': {
+            'targetBucket': uls_name,
+            'region': region,
+            'secret': {
+                'name': cld_mgr.aws_client.secret.name
+            },
+        }
+    }
     create_resource(**bs_data)
 
 
@@ -280,11 +287,12 @@ def cli_create_aws_backingstore(mcg_obj_session, cld_mgr, backingstore_name, uls
         region (str): which region to create backingstore (should be the same as uls)
 
     """
-    mcg_obj_session.exec_mcg_cmd(f'backingstore create aws-s3 {backingstore_name} '
-                                 f'--access-key {cld_mgr.aws_client.access_key} '
-                                 f'--secret-key {cld_mgr.aws_client.secret_key} '
-                                 f'--target-bucket {uls_name} --region {region}'
-                                 )
+    mcg_obj_session.exec_mcg_cmd(
+        f'backingstore create aws-s3 {backingstore_name} '
+        f'--access-key {cld_mgr.aws_client.access_key} '
+        f'--secret-key {cld_mgr.aws_client.secret_key} '
+        f'--target-bucket {uls_name} --region {region}'
+    )
 
 
 def oc_create_google_backingstore(cld_mgr, backingstore_name, uls_name, region):
@@ -295,12 +303,38 @@ def cli_create_google_backingstore(cld_mgr, backingstore_name, uls_name, region)
     pass
 
 
-def oc_create_azure_backingstore(cld_mgr, backingstore_name, uls_name, region):
-    pass
+def oc_create_azure_backingstore(cld_mgr, backingstore_name, uls_name):
+    bs_data = templating.load_yaml(constants.MCG_BACKINGSTORE_YAML)
+    bs_data['metadata']['name'] = backingstore_name
+    bs_data['spec'] = {
+        'type': constants.BACKINGSTORE_TYPE_AZURE,
+        'azureBlob': {
+            'targetBlobContainer': uls_name,
+            'secret': {
+                'name': cld_mgr.azure_client.secret.name
+            },
+        }
+    }
+    create_resource(**bs_data)
 
 
-def cli_create_azure_backingstore(cld_mgr, backingstore_name, uls_name, region):
-    pass
+def cli_create_azure_backingstore(mcg_obj_session, cld_mgr, backingstore_name, uls_name):
+    """
+    Create a new backingstore with aws underlying storage using noobaa cli command
+
+    Args:
+        cld_mgr (CloudManager): holds secret for backingstore creation
+        backingstore_name (str): backingstore name
+        uls_name (str): underlying storage name
+        region (str): which region to create backingstore (should be the same as uls)
+
+    """
+    mcg_obj_session.exec_mcg_cmd(
+        f'backingstore create azure-blob {backingstore_name} '
+        f'--account-key {cld_mgr.azure_client.credential} '
+        f'--account-name {cld_mgr.azure_client.account_name} '
+        f'--target-blob-container {uls_name}'
+    )
 
 
 def oc_create_s3comp_backingstore(cld_mgr, backingstore_name, uls_name, region):

--- a/ocs_ci/ocs/bucket_utils.py
+++ b/ocs_ci/ocs/bucket_utils.py
@@ -318,7 +318,7 @@ def oc_create_azure_backingstore(cld_mgr, backingstore_name, uls_name):
     create_resource(**bs_data)
 
 
-def cli_create_azure_backingstore(mcg_obj_session, cld_mgr, backingstore_name, uls_name):
+def cli_create_azure_backingstore(mcg_obj_session, cld_mgr, backingstore_name, uls_name, region):
     """
     Create a new backingstore with aws underlying storage using noobaa cli command
 

--- a/ocs_ci/ocs/bucket_utils.py
+++ b/ocs_ci/ocs/bucket_utils.py
@@ -478,9 +478,10 @@ def cli_create_pv_backingstore(mcg_obj, backingstore_name, vol_num, size, storag
         storage_class (str): which storage class to use
 
     """
-    mcg_obj.exec_mcg_cmd(f'backingstore create pv-pool {backingstore_name} --num-volumes '
-                                 f'{vol_num} --pv-size-gb {size} --storage-class {storage_class}'
-                                 )
+    mcg_obj.exec_mcg_cmd(
+        f'backingstore create pv-pool {backingstore_name} --num-volumes '
+        f'{vol_num} --pv-size-gb {size} --storage-class {storage_class}'
+    )
     wait_for_pv_backingstore(backingstore_name, config.ENV_DATA['cluster_namespace'])
 
 

--- a/ocs_ci/ocs/bucket_utils.py
+++ b/ocs_ci/ocs/bucket_utils.py
@@ -364,6 +364,60 @@ def cli_create_azure_backingstore(mcg_obj_session, cld_mgr, backingstore_name, u
     )
 
 
+def oc_create_ibmcos_backingstore(cld_mgr, backingstore_name, uls_name, region):
+    """
+    Create a new backingstore with IBM COS underlying storage using oc create command
+
+    Args:
+        cld_mgr (CloudManager): holds secret for backingstore creation
+        backingstore_name (str): backingstore name
+        uls_name (str): underlying storage name
+        region (str): which region to create backingstore (should be the same as uls)
+
+    """
+    bs_data = templating.load_yaml(constants.MCG_BACKINGSTORE_YAML)
+    bs_data['metadata']['name'] = backingstore_name
+    bs_data['metadata']['namespace'] = config.ENV_DATA['cluster_namespace']
+    bs_data['spec'] = {
+        'type': 'ibm-cos',
+        'ibmCos': {
+            'targetBucket': uls_name,
+            'signatureVersion': 'v2',
+            'endpoint': constants.IBM_COS_GEO_ENDPOINT_TEMPLATE.format(
+                cld_mgr.ibmcos_client.region.lower()
+            ),
+            'secret': {
+                'name': cld_mgr.ibmcos_client.secret.name
+            },
+        }
+    }
+    create_resource(**bs_data)
+
+
+def cli_create_ibmcos_backingstore(mcg_obj_session, cld_mgr, backingstore_name, uls_name, region):
+    """
+    Create a new backingstore with IBM COS underlying storage using a NooBaa CLI command
+
+    Args:
+        cld_mgr (CloudManager): holds secret for backingstore creation
+        backingstore_name (str): backingstore name
+        uls_name (str): underlying storage name
+        region (str): which region to create backingstore (should be the same as uls)
+
+    """
+    mcg_obj_session.exec_mcg_cmd(
+        f'backingstore create ibm-cos {backingstore_name} '
+        f'--access-key {cld_mgr.ibmcos_client.access_key} '
+        f'--secret-key {cld_mgr.ibmcos_client.secret_key} '
+        f"""--endpoint {
+            constants.IBM_COS_GEO_ENDPOINT_TEMPLATE.format(
+                cld_mgr.ibmcos_client.region.lower()
+            )
+        } """
+        f'--target-bucket {uls_name}'
+    )
+
+
 def oc_create_s3comp_backingstore(cld_mgr, backingstore_name, uls_name, region):
     pass
 

--- a/ocs_ci/ocs/bucket_utils.py
+++ b/ocs_ci/ocs/bucket_utils.py
@@ -276,19 +276,19 @@ def oc_create_aws_backingstore(cld_mgr, backingstore_name, uls_name, region):
     create_resource(**bs_data)
 
 
-def cli_create_aws_backingstore(mcg_obj_session, cld_mgr, backingstore_name, uls_name, region):
+def cli_create_aws_backingstore(mcg_obj, cld_mgr, backingstore_name, uls_name, region):
     """
     Create a new backingstore with aws underlying storage using noobaa cli command
 
     Args:
-        mcg_obj_session (MCG): Used for execution for the NooBaa CLI command
+        mcg_obj (MCG): Used for execution for the NooBaa CLI command
         cld_mgr (CloudManager): holds secret for backingstore creation
         backingstore_name (str): backingstore name
         uls_name (str): underlying storage name
         region (str): which region to create backingstore (should be the same as uls)
 
     """
-    mcg_obj_session.exec_mcg_cmd(
+    mcg_obj.exec_mcg_cmd(
         f'backingstore create aws-s3 {backingstore_name} '
         f'--access-key {cld_mgr.aws_client.access_key} '
         f'--secret-key {cld_mgr.aws_client.secret_key} '
@@ -321,19 +321,19 @@ def oc_create_google_backingstore(cld_mgr, backingstore_name, uls_name, region):
     create_resource(**bs_data)
 
 
-def cli_create_google_backingstore(mcg_obj_session, cld_mgr, backingstore_name, uls_name, region):
+def cli_create_google_backingstore(mcg_obj, cld_mgr, backingstore_name, uls_name, region):
     """
     Create a new backingstore with GCP underlying storage using a NooBaa CLI command
 
     Args:
-        mcg_obj_session (MCG): Used for execution for the NooBaa CLI command
+        mcg_obj (MCG): Used for execution for the NooBaa CLI command
         cld_mgr (CloudManager): holds secret for backingstore creation
         backingstore_name (str): backingstore name
         uls_name (str): underlying storage name
         region (str): which region to create backingstore (should be the same as uls)
 
     """
-    mcg_obj_session.exec_mcg_cmd(
+    mcg_obj.exec_mcg_cmd(
         f'backingstore create google-cloud-storage {backingstore_name} '
         f'--private-key-json-file {constants.GOOGLE_CREDS_JSON_PATH} '
         f'--target-bucket {uls_name}'
@@ -365,7 +365,7 @@ def oc_create_azure_backingstore(cld_mgr, backingstore_name, uls_name, region):
     create_resource(**bs_data)
 
 
-def cli_create_azure_backingstore(mcg_obj_session, cld_mgr, backingstore_name, uls_name, region):
+def cli_create_azure_backingstore(mcg_obj, cld_mgr, backingstore_name, uls_name, region):
     """
     Create a new backingstore with aws underlying storage using noobaa cli command
 
@@ -376,7 +376,7 @@ def cli_create_azure_backingstore(mcg_obj_session, cld_mgr, backingstore_name, u
         region (str): which region to create backingstore (should be the same as uls)
 
     """
-    mcg_obj_session.exec_mcg_cmd(
+    mcg_obj.exec_mcg_cmd(
         f'backingstore create azure-blob {backingstore_name} '
         f'--account-key {cld_mgr.azure_client.credential} '
         f'--account-name {cld_mgr.azure_client.account_name} '
@@ -414,7 +414,7 @@ def oc_create_ibmcos_backingstore(cld_mgr, backingstore_name, uls_name, region):
     create_resource(**bs_data)
 
 
-def cli_create_ibmcos_backingstore(mcg_obj_session, cld_mgr, backingstore_name, uls_name, region):
+def cli_create_ibmcos_backingstore(mcg_obj, cld_mgr, backingstore_name, uls_name, region):
     """
     Create a new backingstore with IBM COS underlying storage using a NooBaa CLI command
 
@@ -425,7 +425,7 @@ def cli_create_ibmcos_backingstore(mcg_obj_session, cld_mgr, backingstore_name, 
         region (str): which region to create backingstore (should be the same as uls)
 
     """
-    mcg_obj_session.exec_mcg_cmd(
+    mcg_obj.exec_mcg_cmd(
         f'backingstore create ibm-cos {backingstore_name} '
         f'--access-key {cld_mgr.ibmcos_client.access_key} '
         f'--secret-key {cld_mgr.ibmcos_client.secret_key} '
@@ -467,7 +467,7 @@ def oc_create_pv_backingstore(backingstore_name, vol_num, size, storage_class):
     wait_for_pv_backingstore(backingstore_name, config.ENV_DATA['cluster_namespace'])
 
 
-def cli_create_pv_backingstore(mcg_obj_session, backingstore_name, vol_num, size, storage_class):
+def cli_create_pv_backingstore(mcg_obj, backingstore_name, vol_num, size, storage_class):
     """
     Create a new backingstore with pv underlying storage using noobaa cli command
 
@@ -478,7 +478,7 @@ def cli_create_pv_backingstore(mcg_obj_session, backingstore_name, vol_num, size
         storage_class (str): which storage class to use
 
     """
-    mcg_obj_session.exec_mcg_cmd(f'backingstore create pv-pool {backingstore_name} --num-volumes '
+    mcg_obj.exec_mcg_cmd(f'backingstore create pv-pool {backingstore_name} --num-volumes '
                                  f'{vol_num} --pv-size-gb {size} --storage-class {storage_class}'
                                  )
     wait_for_pv_backingstore(backingstore_name, config.ENV_DATA['cluster_namespace'])

--- a/ocs_ci/ocs/bucket_utils.py
+++ b/ocs_ci/ocs/bucket_utils.py
@@ -281,6 +281,7 @@ def cli_create_aws_backingstore(mcg_obj_session, cld_mgr, backingstore_name, uls
     Create a new backingstore with aws underlying storage using noobaa cli command
 
     Args:
+        mcg_obj_session (MCG): Used for execution for the NooBaa CLI command
         cld_mgr (CloudManager): holds secret for backingstore creation
         backingstore_name (str): backingstore name
         uls_name (str): underlying storage name
@@ -296,11 +297,37 @@ def cli_create_aws_backingstore(mcg_obj_session, cld_mgr, backingstore_name, uls
 
 
 def oc_create_google_backingstore(cld_mgr, backingstore_name, uls_name, region):
-    pass
+    bs_data = templating.load_yaml(constants.MCG_BACKINGSTORE_YAML)
+    bs_data['metadata']['name'] = backingstore_name
+    bs_data['spec'] = {
+        'type': constants.BACKINGSTORE_TYPE_GOOGLE,
+        'googleCloudStorage': {
+            'targetBucket': uls_name,
+            'secret': {
+                'name': cld_mgr.gcp_client.secret.name
+            },
+        }
+    }
+    create_resource(**bs_data)
 
 
-def cli_create_google_backingstore(cld_mgr, backingstore_name, uls_name, region):
-    pass
+def cli_create_google_backingstore(mcg_obj_session, cld_mgr, backingstore_name, uls_name, region):
+    """
+    Create a new backingstore with GCP underlying storage using a NooBaa CLI command
+
+    Args:
+        mcg_obj_session (MCG): Used for execution for the NooBaa CLI command
+        cld_mgr (CloudManager): holds secret for backingstore creation
+        backingstore_name (str): backingstore name
+        uls_name (str): underlying storage name
+        region (str): which region to create backingstore (should be the same as uls)
+
+    """
+    mcg_obj_session.exec_mcg_cmd(
+        f'backingstore create google-cloud-storage {backingstore_name} '
+        f'--private-key-json-file {constants.GOOGLE_CREDS_JSON_PATH} '
+        f'--target-bucket {uls_name}'
+    )
 
 
 def oc_create_azure_backingstore(cld_mgr, backingstore_name, uls_name):

--- a/ocs_ci/ocs/bucket_utils.py
+++ b/ocs_ci/ocs/bucket_utils.py
@@ -330,7 +330,7 @@ def cli_create_google_backingstore(mcg_obj_session, cld_mgr, backingstore_name, 
     )
 
 
-def oc_create_azure_backingstore(cld_mgr, backingstore_name, uls_name):
+def oc_create_azure_backingstore(cld_mgr, backingstore_name, uls_name, region):
     bs_data = templating.load_yaml(constants.MCG_BACKINGSTORE_YAML)
     bs_data['metadata']['name'] = backingstore_name
     bs_data['spec'] = {

--- a/ocs_ci/ocs/constants.py
+++ b/ocs_ci/ocs/constants.py
@@ -253,6 +253,7 @@ HOSTNAME_LABEL = "kubernetes.io/hostname"
 # Auth Yaml
 OCSCI_DATA_BUCKET = 'ocs-ci-data'
 AUTHYAML = 'auth.yaml'
+GOOGLE_CREDS_JSON_PATH = os.path.join(DATA_DIR, "google_creds.json")
 
 # OBJ File representing serialized data
 NODE_OBJ_FILE = "node_file.objs"
@@ -1177,6 +1178,7 @@ AZURE_BLOB_ENDPOINT_SUFFIX = 'https://{}.blob.core.windows.net'
 BACKINGSTORE_TYPE_AWS = 'aws-s3'
 BACKINGSTORE_TYPE_AZURE = 'azure-blob'
 BACKINGSTORE_TYPE_S3_COMP = 's3-compatible'
+BACKINGSTORE_TYPE_GOOGLE = 'google-cloud-storage'
 
 # Squads assignment
 # Tests are assigned to Squads based on patterns matching test path.

--- a/ocs_ci/ocs/constants.py
+++ b/ocs_ci/ocs/constants.py
@@ -1172,7 +1172,8 @@ MCG_NS_AWS_CONNECTION = 'aws_connection'
 
 # Cloud provider default endpoints
 # Upon use, utilize .format() to replace the curly braces where necessary
-AZURE_BLOB_ENDPOINT_SUFFIX = 'https://{}.blob.core.windows.net'
+AZURE_BLOB_ENDPOINT_TEMPLATE = 'https://{}.blob.core.windows.net'
+IBM_COS_GEO_ENDPOINT_TEMPLATE = 'https://s3.{}.cloud-object-storage.appdomain.cloud'
 
 # NooBaa backingstore types
 BACKINGSTORE_TYPE_AWS = 'aws-s3'

--- a/ocs_ci/ocs/constants.py
+++ b/ocs_ci/ocs/constants.py
@@ -1169,6 +1169,15 @@ MCG_NS_RESOURCE = 'ns_resource'
 MCG_NS_BUCKET = 'ns-bucket'
 MCG_NS_AWS_CONNECTION = 'aws_connection'
 
+# Cloud provider default endpoints
+# Upon use, utilize .format() to replace the curly braces where necessary
+AZURE_BLOB_ENDPOINT_SUFFIX = 'https://{}.blob.core.windows.net'
+
+# NooBaa backingstore types
+BACKINGSTORE_TYPE_AWS = 'aws-s3'
+BACKINGSTORE_TYPE_AZURE = 'azure-blob'
+BACKINGSTORE_TYPE_S3_COMP = 's3-compatible'
+
 # Squads assignment
 # Tests are assigned to Squads based on patterns matching test path.
 # For example: In case following test fails:

--- a/ocs_ci/ocs/resources/backingstore.py
+++ b/ocs_ci/ocs/resources/backingstore.py
@@ -39,7 +39,7 @@ class BackingStore():
         # Todo: implement deletion assertion
 
 
-def backingstore_factory(request, cld_mgr, cloud_uls_factory):
+def backingstore_factory(request, cld_mgr, cloud_uls_factory, mcg_obj):
     """
     Create a Backing Store factory.
     Calling this fixture creates a new Backing Store(s).
@@ -131,9 +131,14 @@ def backingstore_factory(request, cld_mgr, cloud_uls_factory):
                                 uls_name=uls_name
                             )
                         )
-                        cmdMap[method.lower()][cloud.lower()](
-                            cld_mgr, backingstore_name, uls_name, region
-                        )
+                        if method.lower() == 'cli':
+                            cmdMap[method.lower()][cloud.lower()](
+                                mcg_obj, cld_mgr, backingstore_name, uls_name, region
+                            )
+                        elif method.lower() == 'oc':
+                            cmdMap[method.lower()][cloud.lower()](
+                                cld_mgr, backingstore_name, uls_name, region
+                            )
                         # Todo: Raise an exception in case the BS wasn't created
 
         return created_backingstores

--- a/ocs_ci/ocs/resources/backingstore.py
+++ b/ocs_ci/ocs/resources/backingstore.py
@@ -2,9 +2,9 @@ import logging
 
 from ocs_ci.ocs.bucket_utils import (
     oc_create_aws_backingstore, oc_create_google_backingstore, oc_create_azure_backingstore,
-    oc_create_s3comp_backingstore, oc_create_pv_backingstore, oc_create_ibmcos_backingstore,
-    cli_create_google_backingstore, cli_create_azure_backingstore, cli_create_s3comp_backingstore,
-    cli_create_pv_backingstore, cli_create_ibmcos_backingstore, cli_create_aws_backingstore
+    oc_create_pv_backingstore, oc_create_ibmcos_backingstore, cli_create_google_backingstore,
+    cli_create_azure_backingstore, cli_create_pv_backingstore, cli_create_ibmcos_backingstore,
+    cli_create_aws_backingstore
 )
 from ocs_ci.ocs.ocp import OCP
 from ocs_ci.framework import config

--- a/ocs_ci/ocs/resources/backingstore.py
+++ b/ocs_ci/ocs/resources/backingstore.py
@@ -60,14 +60,14 @@ def backingstore_factory(request, cld_mgr, cloud_uls_factory, mcg_obj):
     cmdMap = {
         'oc': {
             'aws': oc_create_aws_backingstore,
-            'google': oc_create_google_backingstore,
+            'gcp': oc_create_google_backingstore,
             'azure': oc_create_azure_backingstore,
             'ibmcos': oc_create_s3comp_backingstore,
             'pv': oc_create_pv_backingstore
         },
         'cli': {
             'aws': cli_create_aws_backingstore,
-            'google': cli_create_google_backingstore,
+            'gcp': cli_create_google_backingstore,
             'azure': cli_create_azure_backingstore,
             'ibmcos': cli_create_s3comp_backingstore,
             'pv': cli_create_pv_backingstore

--- a/ocs_ci/ocs/resources/backingstore.py
+++ b/ocs_ci/ocs/resources/backingstore.py
@@ -2,9 +2,9 @@ import logging
 
 from ocs_ci.ocs.bucket_utils import (
     oc_create_aws_backingstore, oc_create_google_backingstore, oc_create_azure_backingstore,
-    oc_create_s3comp_backingstore, oc_create_pv_backingstore, cli_create_aws_backingstore,
+    oc_create_s3comp_backingstore, oc_create_pv_backingstore, oc_create_ibmcos_backingstore,
     cli_create_google_backingstore, cli_create_azure_backingstore, cli_create_s3comp_backingstore,
-    cli_create_pv_backingstore
+    cli_create_pv_backingstore, cli_create_ibmcos_backingstore, cli_create_aws_backingstore
 )
 from ocs_ci.ocs.ocp import OCP
 from ocs_ci.framework import config
@@ -62,14 +62,14 @@ def backingstore_factory(request, cld_mgr, cloud_uls_factory, mcg_obj):
             'aws': oc_create_aws_backingstore,
             'gcp': oc_create_google_backingstore,
             'azure': oc_create_azure_backingstore,
-            'ibmcos': oc_create_s3comp_backingstore,
+            'ibmcos': oc_create_ibmcos_backingstore,
             'pv': oc_create_pv_backingstore
         },
         'cli': {
             'aws': cli_create_aws_backingstore,
             'gcp': cli_create_google_backingstore,
             'azure': cli_create_azure_backingstore,
-            'ibmcos': cli_create_s3comp_backingstore,
+            'ibmcos': cli_create_ibmcos_backingstore,
             'pv': cli_create_pv_backingstore
         }
     }
@@ -116,7 +116,7 @@ def backingstore_factory(request, cld_mgr, cloud_uls_factory, mcg_obj):
                         backingstore_name, vol_num, size, storage_class
                     )
                 else:
-                    region = uls_tup[1]
+                    _, region = uls_tup
                     # Todo: Verify that the given cloud has an initialized client
                     uls_dict = cloud_uls_factory({cloud: [uls_tup]})
                     for uls_name in uls_dict[cloud.lower()]:

--- a/ocs_ci/ocs/resources/cloud_manager.py
+++ b/ocs_ci/ocs/resources/cloud_manager.py
@@ -54,7 +54,7 @@ class CloudManager(ABC):
             if cloud_name in cloud_map:
                 if any(value is None for value in cred_dict[cloud_name].values()):
                     logger.warn(
-                        f'{cloud_name} credentials not found in auth.yaml, '
+                        f'{cloud_name} credentials not found '
                         'no client will be instantiated'
                     )
                     setattr(

--- a/ocs_ci/ocs/resources/cloud_manager.py
+++ b/ocs_ci/ocs/resources/cloud_manager.py
@@ -51,15 +51,19 @@ class CloudManager(ABC):
         cred_dict = load_auth_config().get('AUTH', {})
         for cloud_name in cred_dict:
             if cloud_name in cloud_map:
-                try:
-                    setattr(
-                        self, f'{cloud_name.lower()}_client',
-                        cloud_map[cloud_name](auth_dict=cred_dict[cloud_name])
+                if any(value is None for value in cred_dict[cloud_name].values()):
+                    logger.warn(
+                        f'{cloud_name} credentials not found in auth.yaml, '
+                        'no client will be instantiated'
                     )
-                except DefaultCredentialsError:
                     setattr(
                         self, f'{cloud_name.lower()}_client',
                         None
+                    )
+                else:
+                    setattr(
+                        self, f'{cloud_name.lower()}_client',
+                        cloud_map[cloud_name](auth_dict=cred_dict[cloud_name])
                     )
 
 

--- a/ocs_ci/ocs/resources/cloud_manager.py
+++ b/ocs_ci/ocs/resources/cloud_manager.py
@@ -367,7 +367,7 @@ class AzureClient(CloudClient):
                 credential=credential
             )
 
-    def internal_create_uls(self, name):
+    def internal_create_uls(self, name, region):
         """
         Creates the Underlying Storage using the Azure API
 

--- a/ocs_ci/ocs/resources/cloud_manager.py
+++ b/ocs_ci/ocs/resources/cloud_manager.py
@@ -5,6 +5,8 @@ import os
 from abc import ABC, abstractmethod
 from time import sleep
 
+from azure.storage.blob import BlobServiceClient, ContainerClient
+from azure.core.exceptions import ResourceExistsError
 import boto3
 from botocore.exceptions import ClientError
 from google.auth.exceptions import DefaultCredentialsError
@@ -346,21 +348,34 @@ class AzureClient(CloudClient):
 
     """
 
-    def __init__(self, *args, **kwargs):
+    def __init__(
+        self, account_name=None, credential=None,
+        auth_dict=None, *args, **kwargs
+    ):
         super().__init__(*args, **kwargs)
-        # TODO: Implement
-        raise DefaultCredentialsError
+        if auth_dict:
+            account_name = auth_dict.get('STORAGE_ACCOUNT_NAME')
+            credential = auth_dict.get('STORAGE_ACCOUNT_KEY')
+        if account_name and credential:
+            self.account_name = account_name
+            self.credential = credential
+            self.secret = self.create_azure_secret()
 
-    def internal_create_uls(self, name, region=None):
+            account_url = constants.AZURE_BLOB_ENDPOINT_SUFFIX.format(account_name)
+            self.blob_service_client = BlobServiceClient(
+                account_url=account_url,
+                credential=credential
+            )
+
+    def internal_create_uls(self, name):
         """
         Creates the Underlying Storage using the Azure API
 
         Args:
            name (str): The Underlying Storage name to be created
-           region (str): The region to create the Underlying Storage,
 
         """
-        pass
+        self.blob_service_client.get_container_client(name).create_container()
 
     def internal_delete_uls(self, name):
         """
@@ -370,10 +385,33 @@ class AzureClient(CloudClient):
            name (str): The Underlying Storage name to be deleted
 
         """
-        pass
+        self.blob_service_client.get_container_client(name).delete_container()
 
     def get_all_uls_names(self):
-        pass
+        return [
+            container['name'] for container in self.blob_service_client.list_containers()
+        ]
 
     def verify_uls_exists(self, uls_name):
-        pass
+        try:
+            self.blob_service_client.get_container_client(uls_name).get_container_properties()
+            return True
+        except ResourceExistsError:
+            return False
+
+    def create_azure_secret(self):
+        """
+        Create a Kubernetes secret to allow NooBaa to create Azure-based backingstores
+
+        """
+        bs_secret_data = templating.load_yaml(constants.MCG_BACKINGSTORE_SECRET_YAML)
+        bs_secret_data['metadata']['name'] = 'cldmgr-azure-secret'
+        bs_secret_data['metadata']['namespace'] = config.ENV_DATA['cluster_namespace']
+        bs_secret_data['data']['AccountKey'] = base64.urlsafe_b64encode(
+            self.credential.encode('UTF-8')
+        ).decode('ascii')
+        bs_secret_data['data']['AccountName'] = base64.urlsafe_b64encode(
+            self.account_name.encode('UTF-8')
+        ).decode('ascii')
+
+        return create_resource(**bs_secret_data)

--- a/ocs_ci/ocs/resources/cloud_manager.py
+++ b/ocs_ci/ocs/resources/cloud_manager.py
@@ -351,7 +351,7 @@ class GoogleClient(CloudClient):
                 sleep(3)
 
     def get_all_uls_names(self):
-        return self.client.list_buckets()
+        return {bucketname for bucketname in self.client.list_buckets()}
 
     def verify_uls_exists(self, uls_name):
         try:
@@ -421,9 +421,9 @@ class AzureClient(CloudClient):
         self.blob_service_client.get_container_client(name).delete_container()
 
     def get_all_uls_names(self):
-        return [
+        return {
             container['name'] for container in self.blob_service_client.list_containers()
-        ]
+        }
 
     def verify_uls_exists(self, uls_name):
         try:

--- a/ocs_ci/ocs/resources/cloud_manager.py
+++ b/ocs_ci/ocs/resources/cloud_manager.py
@@ -16,6 +16,7 @@ from google.oauth2 import service_account
 
 from ocs_ci.framework import config
 from ocs_ci.ocs import constants
+from ocs_ci.ocs.exceptions import TimeoutExpiredError
 from ocs_ci.utility import templating
 from ocs_ci.utility.aws import update_config_from_s3
 from ocs_ci.utility.utils import TimeoutSampler, load_auth_config
@@ -34,10 +35,13 @@ class CloudManager(ABC):
             'AWS': S3Client,
             'GCP': GoogleClient,
             'AZURE': AzureClient,
-            # TODO: Implement - 'IBMCOS': S3Client
+            'IBMCOS': S3Client
         }
         try:
-            logger.info('Trying to load credentials from ocs-ci-data')
+            logger.info(
+                'Trying to load credentials from ocs-ci-data. '
+                'This flow is only relevant when running under OCS-QE environments.'
+            )
             cred_dict = update_config_from_s3().get('AUTH')
         except AttributeError:
             logger.warn(
@@ -107,39 +111,27 @@ class S3Client(CloudClient):
     Implementation of a S3 Client using the S3 API
 
     """
-    def __init__(self, key_id=None, access_key=None, endpoint="https://s3.amazonaws.com",
-                 verify=True, auth_dict=None, *args, **kwargs):
+    def __init__(
+        self, auth_dict, verify=True,
+        endpoint="https://s3.amazonaws.com", *args, **kwargs
+    ):
         super().__init__(*args, **kwargs)
-        if auth_dict:
-            key_id = auth_dict.get('AWS_ACCESS_KEY_ID')
-            access_key = auth_dict.get('AWS_SECRET_ACCESS_KEY')
-            endpoint = endpoint or auth_dict.get('ENDPOINT')
-            s3_comp = auth_dict.get('S3_COMP')
-        if key_id and access_key:
-            self.client = boto3.resource(
-                's3', verify=verify, endpoint_url=endpoint,
-                aws_access_key_id=key_id,
-                aws_secret_access_key=access_key
-            )
-            self.access_key = key_id
-            self.secret_key = access_key
-        elif s3_comp:
-            logger.warning("Did not find S3 compatible credentials")
-            raise DefaultCredentialsError
-        else:
-            try:
-                self.client = boto3.resource('s3', endpoint_url=endpoint)
-                # create a secret for the underlying storage to use
-                session = boto3.Session()
-                # Retrieving the credentials of the existing session
-                credentials = session.get_credentials().get_frozen_credentials()
-                self.access_key = credentials.access_key
-                self.secret_key = credentials.secret_key
-            except AttributeError:
-                logger.warning("Failed to find default AWS credentials")
-                raise DefaultCredentialsError
 
-        self.secret = self.create_aws_secret()
+        secret_prefix = auth_dict.get('SECRET_PREFIX', 'AWS')
+        key_id = auth_dict.get(f'{secret_prefix}_ACCESS_KEY_ID')
+        access_key = auth_dict.get(f'{secret_prefix}_SECRET_ACCESS_KEY')
+        self.endpoint = auth_dict.get('ENDPOINT') or endpoint
+        self.region = auth_dict.get('REGION')
+        self.access_key = key_id
+        self.secret_key = access_key
+
+        self.client = boto3.resource(
+            's3', verify=verify, endpoint_url=self.endpoint,
+            aws_access_key_id=self.access_key,
+            aws_secret_access_key=self.secret_key
+        )
+
+        self.secret = self.create_s3_secret(secret_prefix)
 
     def internal_create_uls(self, name, region=None):
         """
@@ -147,9 +139,11 @@ class S3Client(CloudClient):
 
         Args:
            name (str): The Underlying Storage name to be created
-           region (str): The region to create the Underlying Storage, if none will create at
-           `us-east-1` IMPORTANT!!! note that `us-east-1` will cause an error if used since it is
-           the default region for aws
+           region (str): The region to create the Underlying Storage,
+           if none will be created on `us-east-1`
+           **IMPORTANT**
+           Passing `us-east-1` as the region will cause an error if used since it is
+           the default region for AWS
 
         """
         if region is None:
@@ -170,7 +164,45 @@ class S3Client(CloudClient):
            name (str): The Underlying Storage name to be deleted
 
         """
-        assert self.exec_uls_deletion(name)
+        def _exec_uls_deletion(self, name):
+            """
+            Try to delete Underlying Storage by name if exists
+
+            Args:
+                name (str): the Underlying Storage name
+
+            Returns:
+                bool: True if deleted successfully
+
+            """
+            if self.verify_uls_exists(name):
+                try:
+                    if 'aws' in name:
+                        self.client.meta.client.delete_bucket_policy(
+                            Bucket=name
+                        )
+                    self.client.Bucket(name).objects.all().delete()
+                    self.client.Bucket(name).delete()
+                    return True
+                except ClientError:
+                    logger.info(f'Deletion of Underlying Storage {name} failed.')
+                    return False
+            else:
+                logger.warning(f'Underlying Storage {name} does not exist, and was not deleted.')
+                return True
+
+        try:
+            for deletion_result in TimeoutSampler(60, 5, _exec_uls_deletion, self, name):
+                if deletion_result:
+                    logger.info('ULS deleted.')
+                    break
+
+        except TimeoutExpiredError:
+            logger.error(
+                'Failed to delete ULS.'
+            )
+            assert False
+
         # Todo: rename client to resource (or find an alternative)
         sample = TimeoutSampler(
             timeout=180, sleep=15, func=self.verify_uls_exists,
@@ -186,43 +218,21 @@ class S3Client(CloudClient):
         else:
             logger.info(f'Underlying Storage {name} deleted successfully.')
 
-    def exec_uls_deletion(self, name):
-        """
-        Try to delete Underlying Storage by name if exists
-
-        Args:
-            name (str): the Underlying Storage name
-
-        Returns:
-            bool: True if deleted successfully
-
-        """
-        if self.verify_uls_exists(name):
-            try:
-                self.client.meta.client.delete_bucket_policy(
-                    Bucket=name
-                )
-                self.client.Bucket(name).objects.all().delete()
-                self.client.Bucket(name).delete()
-                return True
-            except ClientError:
-                logger.info(f'Deletion of Underlying Storage {name} failed.')
-                return False
-        else:
-            logger.warning(f'Underlying Storage {name} does not exist, and was not deleted.')
-            return True
-
     def get_all_uls_names(self):
+        """
+        Returns a set containing all the bucket names that the client has access to
+
+        """
         return {bucket.name for bucket in self.client.buckets.all()}
 
     def verify_uls_exists(self, uls_name):
         """
-       Verifies whether a Underlying Storage with the given uls_name exists
+        Verifies whether a Underlying Storage with the given uls_name exists
 
-       Args:
+        Args:
            uls_name (str): The Underlying Storage name to be verified
 
-       Returns:
+        Returns:
              bool: True if Underlying Storage exists, False otherwise
 
         """
@@ -275,14 +285,20 @@ class S3Client(CloudClient):
                 Bucket=aws_bucket_name
             )
 
-    def create_aws_secret(self):
+    def create_s3_secret(self, secret_prefix):
+        """
+        Create a Kubernetes secret to allow NooBaa to create AWS-based backingstores
+
+        """
         bs_secret_data = templating.load_yaml(constants.MCG_BACKINGSTORE_SECRET_YAML)
-        bs_secret_data['metadata']['name'] = 'cldmgr-aws-secret'
+        secret_name_prefix = secret_prefix.lower()
+        secret_name_prefix = secret_name_prefix.replace('_', '-')
+        bs_secret_data['metadata']['name'] = f'cldmgr-{secret_name_prefix}-secret'
         bs_secret_data['metadata']['namespace'] = config.ENV_DATA['cluster_namespace']
-        bs_secret_data['data']['AWS_ACCESS_KEY_ID'] = base64.urlsafe_b64encode(
+        bs_secret_data['data'][f'{secret_prefix}_ACCESS_KEY_ID'] = base64.urlsafe_b64encode(
             self.access_key.encode('UTF-8')
         ).decode('ascii')
-        bs_secret_data['data']['AWS_SECRET_ACCESS_KEY'] = base64.urlsafe_b64encode(
+        bs_secret_data['data'][f'{secret_prefix}_SECRET_ACCESS_KEY'] = base64.urlsafe_b64encode(
             self.secret_key.encode('UTF-8')
         ).decode('ascii')
 
@@ -351,9 +367,23 @@ class GoogleClient(CloudClient):
                 sleep(3)
 
     def get_all_uls_names(self):
+        """
+        Returns a set containing all the bucket names that the client has access to
+
+        """
         return {bucket.id for bucket in self.client.list_buckets()}
 
     def verify_uls_exists(self, uls_name):
+        """
+        Verifies whether a Underlying Storage with the given uls_name exists
+
+        Args:
+           uls_name (str): The Underlying Storage name to be verified
+
+        Returns:
+             bool: True if Underlying Storage exists, False otherwise
+
+        """
         try:
             self.client.get_bucket(uls_name)
             return True
@@ -394,7 +424,7 @@ class AzureClient(CloudClient):
             self.credential = credential
             self.secret = self.create_azure_secret()
 
-            account_url = constants.AZURE_BLOB_ENDPOINT_SUFFIX.format(account_name)
+            account_url = constants.AZURE_BLOB_ENDPOINT_TEMPLATE.format(account_name)
             self.blob_service_client = BlobServiceClient(
                 account_url=account_url,
                 credential=credential
@@ -421,11 +451,25 @@ class AzureClient(CloudClient):
         self.blob_service_client.get_container_client(name).delete_container()
 
     def get_all_uls_names(self):
+        """
+        Returns a set containing all the container names that the client has access to
+
+        """
         return {
             container['name'] for container in self.blob_service_client.list_containers()
         }
 
     def verify_uls_exists(self, uls_name):
+        """
+        Verifies whether a Underlying Storage with the given uls_name exists
+
+        Args:
+           uls_name (str): The Underlying Storage name to be verified
+
+        Returns:
+             bool: True if Underlying Storage exists, False otherwise
+
+        """
         try:
             self.blob_service_client.get_container_client(uls_name).get_container_properties()
             return True

--- a/ocs_ci/ocs/resources/cloud_manager.py
+++ b/ocs_ci/ocs/resources/cloud_manager.py
@@ -5,12 +5,15 @@ import os
 from abc import ABC, abstractmethod
 from time import sleep
 
-from azure.storage.blob import BlobServiceClient, ContainerClient
-from azure.core.exceptions import ResourceExistsError
 import boto3
+import google.api_core.exceptions as GoogleExceptions
+from azure.core.exceptions import ResourceExistsError
+from azure.storage.blob import BlobServiceClient
 from botocore.exceptions import ClientError
 from google.auth.exceptions import DefaultCredentialsError
-from google.cloud import storage
+from google.cloud.storage import Client as GCPStorageClient
+from google.cloud.storage.bucket import Bucket as GCPBucket
+from google.oauth2 import service_account
 
 from ocs_ci.framework import config
 from ocs_ci.ocs import constants
@@ -30,7 +33,7 @@ class CloudManager(ABC):
     def __init__(self):
         cloud_map = {
             'AWS': S3Client,
-            'GOOGLE': GoogleClient,
+            'GCP': GoogleClient,
             'AZURE': AzureClient,
             # TODO: Implement - 'IBMCOS': S3Client
         }
@@ -42,7 +45,7 @@ class CloudManager(ABC):
                 'Failed to load credentials from ocs-ci-data. '
                 'Loading from local auth.yaml'
             )
-            cred_dict = load_auth_config().get('AUTH', {})
+        cred_dict = load_auth_config().get('AUTH', {})
         for cloud_name in cred_dict:
             if cloud_name in cloud_map:
                 try:
@@ -293,12 +296,25 @@ class GoogleClient(CloudClient):
 
     """
 
-    def __init__(self, creds=None, *args, **kwargs):
+    def __init__(self, auth_dict, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        if creds:
-            os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = creds
+        self.cred_dict_string = base64.b64decode(
+            auth_dict.get('CREDENTIALS_JSON_BASE64')
+        ).decode('utf-8')
+        cred_dict = json.loads(self.cred_dict_string)
+        credentials = service_account.Credentials.from_service_account_info(
+            cred_dict
+        )
+
+        with open(constants.GOOGLE_CREDS_JSON_PATH, 'w') as cred_dump:
+            cred_dump.write(self.cred_dict_string)
+
+        self.secret = self.create_gcp_secret()
+
         try:
-            self.client = storage.Client()
+            self.client = GCPStorageClient(
+                project=cred_dict['project_id'], credentials=credentials
+            )
         except DefaultCredentialsError:
             raise
 
@@ -327,7 +343,7 @@ class GoogleClient(CloudClient):
         # Todo: Replace with a TimeoutSampler
         for _ in range(10):
             try:
-                bucket = self.client.get_bucket(name)
+                bucket = GCPBucket(client=self.client, name=name)
                 bucket.delete_blobs(bucket.list_blobs())
                 bucket.delete()
                 break
@@ -336,10 +352,28 @@ class GoogleClient(CloudClient):
                 sleep(3)
 
     def get_all_uls_names(self):
-        pass
+        return self.client.list_buckets()
 
     def verify_uls_exists(self, uls_name):
-        pass
+        try:
+            self.client.get_bucket(uls_name)
+            return True
+        except GoogleExceptions.NotFound:
+            return False
+
+    def create_gcp_secret(self):
+            """
+            Create a Kubernetes secret to allow NooBaa to create Google-based backingstores
+
+            """
+            bs_secret_data = templating.load_yaml(constants.MCG_BACKINGSTORE_SECRET_YAML)
+            bs_secret_data['metadata']['name'] = 'cldmgr-gcp-secret'
+            bs_secret_data['metadata']['namespace'] = config.ENV_DATA['cluster_namespace']
+            bs_secret_data['data']['GoogleServiceAccountPrivateKeyJson'] = base64.urlsafe_b64encode(
+                self.cred_dict_string.encode('UTF-8')
+            ).decode('ascii')
+
+            return create_resource(**bs_secret_data)
 
 
 class AzureClient(CloudClient):

--- a/ocs_ci/ocs/resources/cloud_manager.py
+++ b/ocs_ci/ocs/resources/cloud_manager.py
@@ -1,7 +1,6 @@
 import base64
 import json
 import logging
-import os
 from abc import ABC, abstractmethod
 from time import sleep
 
@@ -362,18 +361,18 @@ class GoogleClient(CloudClient):
             return False
 
     def create_gcp_secret(self):
-            """
-            Create a Kubernetes secret to allow NooBaa to create Google-based backingstores
+        """
+        Create a Kubernetes secret to allow NooBaa to create Google-based backingstores
 
-            """
-            bs_secret_data = templating.load_yaml(constants.MCG_BACKINGSTORE_SECRET_YAML)
-            bs_secret_data['metadata']['name'] = 'cldmgr-gcp-secret'
-            bs_secret_data['metadata']['namespace'] = config.ENV_DATA['cluster_namespace']
-            bs_secret_data['data']['GoogleServiceAccountPrivateKeyJson'] = base64.urlsafe_b64encode(
-                self.cred_dict_string.encode('UTF-8')
-            ).decode('ascii')
+        """
+        bs_secret_data = templating.load_yaml(constants.MCG_BACKINGSTORE_SECRET_YAML)
+        bs_secret_data['metadata']['name'] = 'cldmgr-gcp-secret'
+        bs_secret_data['metadata']['namespace'] = config.ENV_DATA['cluster_namespace']
+        bs_secret_data['data']['GoogleServiceAccountPrivateKeyJson'] = base64.urlsafe_b64encode(
+            self.cred_dict_string.encode('UTF-8')
+        ).decode('ascii')
 
-            return create_resource(**bs_secret_data)
+        return create_resource(**bs_secret_data)
 
 
 class AzureClient(CloudClient):

--- a/ocs_ci/ocs/resources/cloud_manager.py
+++ b/ocs_ci/ocs/resources/cloud_manager.py
@@ -351,7 +351,7 @@ class GoogleClient(CloudClient):
                 sleep(3)
 
     def get_all_uls_names(self):
-        return {bucketname for bucketname in self.client.list_buckets()}
+        return {bucket.id for bucket in self.client.list_buckets()}
 
     def verify_uls_exists(self, uls_name):
         try:

--- a/ocs_ci/ocs/resources/cloud_manager.py
+++ b/ocs_ci/ocs/resources/cloud_manager.py
@@ -48,7 +48,8 @@ class CloudManager(ABC):
                 'Failed to load credentials from ocs-ci-data. '
                 'Loading from local auth.yaml'
             )
-        cred_dict = load_auth_config().get('AUTH', {})
+            cred_dict = load_auth_config().get('AUTH', {})
+
         for cloud_name in cred_dict:
             if cloud_name in cloud_map:
                 if any(value is None for value in cred_dict[cloud_name].values()):

--- a/ocs_ci/ocs/resources/cloud_manager.py
+++ b/ocs_ci/ocs/resources/cloud_manager.py
@@ -181,6 +181,8 @@ class S3Client(CloudClient):
             """
             if self.verify_uls_exists(name):
                 try:
+                    # TODO: Check why bucket policy deletion fails on IBM COS
+                    # when bucket have no policy set
                     if 'aws' in name:
                         self.client.meta.client.delete_bucket_policy(
                             Bucket=name

--- a/ocs_ci/ocs/resources/cloud_uls.py
+++ b/ocs_ci/ocs/resources/cloud_uls.py
@@ -70,11 +70,11 @@ def cloud_uls_factory(request, cld_mgr):
                     )
                     try:
                         ulsMap[cloud.lower()].create_uls(uls_name, region)
-                    except AttributeError:
-                        raise(
+                    except AttributeError as e:
+                        raise Exception(
                             f'{cloud} was initialized as None, '
                             'please verify the needed credentials were set in auth.yaml'
-                        )
+                        ).with_traceback(e.__traceback__)
                     all_created_uls[cloud].add(uls_name)
                     current_call_created_uls[cloud.lower()].add(uls_name)
 

--- a/ocs_ci/ocs/resources/cloud_uls.py
+++ b/ocs_ci/ocs/resources/cloud_uls.py
@@ -30,7 +30,7 @@ def cloud_uls_factory(request, cld_mgr):
         'aws': cld_mgr.aws_client,
         'gcp': cld_mgr.gcp_client,
         'azure': cld_mgr.azure_client,
-        # TODO: Implement - 'ibmcos': cld_mgr.ibmcos_client
+        'ibmcos': cld_mgr.ibmcos_client
     }
 
     def _create_uls(uls_dict):
@@ -63,9 +63,8 @@ def cloud_uls_factory(request, cld_mgr):
                     f'available types: {", ".join(ulsMap.keys())}'
                 )
             log.info(f'Creating uls for cloud {cloud.lower()}')
-            for tup in params:
-                amount, region = tup
-                for i in range(amount):
+            for amount, region in params:
+                for _ in range(amount):
                     uls_name = create_unique_resource_name(
                         resource_description='uls', resource_type=cloud.lower()
                     )

--- a/ocs_ci/ocs/resources/cloud_uls.py
+++ b/ocs_ci/ocs/resources/cloud_uls.py
@@ -25,13 +25,19 @@ def cloud_uls_factory(request, cld_mgr):
         'azure': set(),
         'ibmcos': set()
     }
-
-    ulsMap = {
-        'aws': cld_mgr.aws_client,
-        'gcp': cld_mgr.gcp_client,
-        'azure': cld_mgr.azure_client,
-        'ibmcos': cld_mgr.ibmcos_client
-    }
+    try:
+        ulsMap = {
+            'aws': cld_mgr.aws_client,
+            'gcp': cld_mgr.gcp_client,
+            'azure': cld_mgr.azure_client,
+            'ibmcos': cld_mgr.ibmcos_client
+        }
+    except AttributeError as e:
+        raise Exception(
+            '{} was not initialized, '
+            'please verify the needed credentials '
+            'were set in auth.yaml'.format(str(e).split("'")[3])
+        ).with_traceback(e.__traceback__)
 
     def _create_uls(uls_dict):
         """

--- a/ocs_ci/ocs/resources/cloud_uls.py
+++ b/ocs_ci/ocs/resources/cloud_uls.py
@@ -21,14 +21,14 @@ def cloud_uls_factory(request, cld_mgr):
     """
     all_created_uls = {
         'aws': set(),
-        'google': set(),
+        'gcp': set(),
         'azure': set(),
         'ibmcos': set()
     }
 
     ulsMap = {
         'aws': cld_mgr.aws_client,
-        'google': cld_mgr.google_client,
+        'gcp': cld_mgr.gcp_client,
         'azure': cld_mgr.azure_client,
         # TODO: Implement - 'ibmcos': cld_mgr.ibmcos_client
     }
@@ -51,7 +51,7 @@ def cloud_uls_factory(request, cld_mgr):
         """
         current_call_created_uls = {
             'aws': set(),
-            'google': set(),
+            'gcp': set(),
             'azure': set(),
             'ibmcos': set()
         }

--- a/ocs_ci/ocs/resources/cloud_uls.py
+++ b/ocs_ci/ocs/resources/cloud_uls.py
@@ -37,7 +37,7 @@ def cloud_uls_factory(request, cld_mgr):
             '{} was not initialized, '
             'please verify the needed credentials '
             'were set in auth.yaml'.format(str(e).split("'")[3])
-        ).with_traceback(e.__traceback__)
+        )
 
     def _create_uls(uls_dict):
         """

--- a/ocs_ci/ocs/resources/cloud_uls.py
+++ b/ocs_ci/ocs/resources/cloud_uls.py
@@ -68,7 +68,13 @@ def cloud_uls_factory(request, cld_mgr):
                     uls_name = create_unique_resource_name(
                         resource_description='uls', resource_type=cloud.lower()
                     )
-                    ulsMap[cloud.lower()].create_uls(uls_name, region)
+                    try:
+                        ulsMap[cloud.lower()].create_uls(uls_name, region)
+                    except AttributeError:
+                        raise(
+                            f'{cloud} was initialized as None, '
+                            'please verify the needed credentials were set in auth.yaml'
+                        )
                     all_created_uls[cloud].add(uls_name)
                     current_call_created_uls[cloud.lower()].add(uls_name)
 

--- a/ocs_ci/templates/mcg/BackingStore.yaml
+++ b/ocs_ci/templates/mcg/BackingStore.yaml
@@ -8,10 +8,3 @@ metadata:
     app: noobaa
   finalizers:
     - noobaa.io/finalizer
-spec:
-  type: aws-s3
-  awsS3:
-    targetBucket: noobaa-aws1
-    region: region
-    secret:
-      name: backing-store-secret

--- a/ocs_ci/templates/mcg/BackingStoreSecret.yaml
+++ b/ocs_ci/templates/mcg/BackingStoreSecret.yaml
@@ -6,5 +6,3 @@ metadata:
   namespace: openshift-storage
 type: Opaque
 data:
-  AWS_ACCESS_KEY_ID: SECRET-ID
-  AWS_SECRET_ACCESS_KEY: SECRET-KEY

--- a/ocs_ci/templates/mcg/BackingStoreSecret.yaml
+++ b/ocs_ci/templates/mcg/BackingStoreSecret.yaml
@@ -6,3 +6,4 @@ metadata:
   namespace: openshift-storage
 type: Opaque
 data:
+  {}

--- a/setup.py
+++ b/setup.py
@@ -54,6 +54,7 @@ setup(
         'azure-mgmt-compute==12.0.0',
         'azure-mgmt-network==10.2.0',
         'azure-mgmt-resource==10.0.0',
+        'azure-storage-blob==12.5.0',
         'msrestazure==0.6.3',
         'python-novaclient==17.1.0',
         'python-cinderclient==7.1.0',

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -2082,7 +2082,7 @@ def multiregion_resources_fixture(request, cld_mgr, mcg_obj):
 @pytest.fixture()
 def multiregion_mirror_setup(
     mcg_obj, multiregion_resources,
-    backingstore_factory_session, bucket_factory_session
+    backingstore_factory, bucket_factory
 ):
     return multiregion_mirror_setup_fixture(
         mcg_obj,

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -2013,7 +2013,7 @@ def mcg_job_factory_session(
     )
 
 
-@pytest.fixture()
+@pytest.fixture(scope='class')
 def backingstore_factory(request, cld_mgr, cloud_uls_factory, mcg_obj):
     """
         Create a Backing Store factory.

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -2013,6 +2013,7 @@ def mcg_job_factory_session(
     )
 
 
+@pytest.fixture()
 def backingstore_factory(request, cld_mgr, cloud_uls_factory, mcg_obj):
     """
         Create a Backing Store factory.
@@ -2079,7 +2080,10 @@ def multiregion_resources_fixture(request, cld_mgr, mcg_obj):
 
 
 @pytest.fixture()
-def multiregion_mirror_setup(mcg_obj, multiregion_resources, backingstore_factory, bucket_factory):
+def multiregion_mirror_setup(
+    mcg_obj, multiregion_resources,
+    backingstore_factory_session, bucket_factory_session
+):
     return multiregion_mirror_setup_fixture(
         mcg_obj,
         multiregion_resources,

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -2013,8 +2013,7 @@ def mcg_job_factory_session(
     )
 
 
-@pytest.fixture(scope='class')
-def backingstore_factory(request, cld_mgr, cloud_uls_factory):
+def backingstore_factory(request, cld_mgr, cloud_uls_factory, mcg_obj):
     """
         Create a Backing Store factory.
         Calling this fixture creates a new Backing Store(s).
@@ -2027,12 +2026,13 @@ def backingstore_factory(request, cld_mgr, cloud_uls_factory):
     return backingstore_factory_implementation(
         request,
         cld_mgr,
-        cloud_uls_factory
+        cloud_uls_factory,
+        mcg_obj
     )
 
 
 @pytest.fixture(scope='session')
-def backingstore_factory_session(request, cld_mgr, cloud_uls_factory_session):
+def backingstore_factory_session(request, cld_mgr, cloud_uls_factory_session, mcg_obj_session):
     """
         Create a Backing Store factory.
         Calling this fixture creates a new Backing Store(s).
@@ -2045,7 +2045,8 @@ def backingstore_factory_session(request, cld_mgr, cloud_uls_factory_session):
     return backingstore_factory_implementation(
         request,
         cld_mgr,
-        cloud_uls_factory_session
+        cloud_uls_factory_session,
+        mcg_obj_session
     )
 
 

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -2013,7 +2013,7 @@ def mcg_job_factory_session(
     )
 
 
-@pytest.fixture(scope='class')
+@pytest.fixture()
 def backingstore_factory(request, cld_mgr, cloud_uls_factory, mcg_obj):
     """
         Create a Backing Store factory.

--- a/tests/manage/mcg/test_multi_region.py
+++ b/tests/manage/mcg/test_multi_region.py
@@ -27,12 +27,14 @@ class TestMultiRegion:
 
     @tier1
     @pytest.mark.polarion_id("OCS-1599")
-    def test_multiregion_bucket_creation(self, mcg_obj, multiregion_mirror_setup):
+    def test_multiregion_bucket_creation(
+        self, mcg_obj, multiregion_mirror_setup_session
+    ):
         """
         Test bucket creation using the S3 SDK
         """
 
-        mirrored_bucket_name = multiregion_mirror_setup[0].name
+        mirrored_bucket_name = multiregion_mirror_setup_session[0].name
         system_bucket, mirror_tier_name, mirror_attached_pools = (None,) * 3
 
         # Make sure that the bucket is up and running
@@ -70,12 +72,14 @@ class TestMultiRegion:
     @bugzilla('1827317')
     @skipif_ocs_version("==4.4")
     @pytest.mark.polarion_id("OCS-1784")
-    def test_multiregion_mirror(self, cld_mgr, mcg_obj, awscli_pod, multiregion_mirror_setup):
+    def test_multiregion_mirror(
+        self, cld_mgr, mcg_obj, awscli_pod, multiregion_mirror_setup_session
+    ):
         """
         Test multi-region bucket creation using the S3 SDK
         """
 
-        bucket, backingstores = multiregion_mirror_setup
+        bucket, backingstores = multiregion_mirror_setup_session
         backingstore1 = backingstores[0]
         backingstore2 = backingstores[1]
 

--- a/tests/manage/mcg/test_multi_region.py
+++ b/tests/manage/mcg/test_multi_region.py
@@ -28,13 +28,13 @@ class TestMultiRegion:
     @tier1
     @pytest.mark.polarion_id("OCS-1599")
     def test_multiregion_bucket_creation(
-        self, mcg_obj, multiregion_mirror_setup_session
+        self, mcg_obj, multiregion_mirror_setup
     ):
         """
         Test bucket creation using the S3 SDK
         """
 
-        mirrored_bucket_name = multiregion_mirror_setup_session[0].name
+        mirrored_bucket_name = multiregion_mirror_setup[0].name
         system_bucket, mirror_tier_name, mirror_attached_pools = (None,) * 3
 
         # Make sure that the bucket is up and running
@@ -73,13 +73,13 @@ class TestMultiRegion:
     @skipif_ocs_version("==4.4")
     @pytest.mark.polarion_id("OCS-1784")
     def test_multiregion_mirror(
-        self, cld_mgr, mcg_obj, awscli_pod, multiregion_mirror_setup_session
+        self, cld_mgr, mcg_obj, awscli_pod, multiregion_mirror_setup
     ):
         """
         Test multi-region bucket creation using the S3 SDK
         """
 
-        bucket, backingstores = multiregion_mirror_setup_session
+        bucket, backingstores = multiregion_mirror_setup
         backingstore1 = backingstores[0]
         backingstore2 = backingstores[1]
 

